### PR TITLE
Refactor testing

### PR DESCRIPTION
**Pull request by ludeeus**
~~Note: This PR currently includes https://github.com/hacs/integration/pull/3566, we should probably merge that PR first~~

~~Needs https://github.com/hacs/integration/pull/3566 for tests to pass~~

### CODE REVIEW

File: tests/common.py [hunk 1]
```diff
@@ -775,6 +775,7 @@ async def json(**kwargs):
                 "X-RateLimit-Remaining": "999",
                 "X-RateLimit-Reset": "999",
                 "Content-Type": "application/json",
+                "Etag": "321",
```
Comment (emontnemery) [inline tests/common.py:778]: Is it OK to always include this here, or should it be more targeted?

Comment (ludeeus) [reply tests/common.py:778]: I think that is fine, if needed to adjust later we can always add an override for it.

File: tests/test_config_flow.py
Comment (emontnemery) [inline tests/test_config_flow.py:1]: Is the option flow not yet tested?

File: tests/test_data_client.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    _result: str | None = None
+
+    response_mocker.add(
+        "https://data-v2.hacs.xyz/integration/repositories.json",
+        response=MockedResponse(exception=exception),
+    )
+
+    try:
+        await hacs.data_client.get_repositories("integration")
+    except Exception as exception:
```
Comment (emontnemery) [inline tests/test_data_client.py:75]: Can we use `pytest.raises`?
```
with pytest.raises(exception, match=result):
```

Same comment for other tests

File: tests/test_config_flow.py
Comment (ludeeus) [reply tests/test_config_flow.py:1]: Probably not, I mostly gave up on the config flow test because of mutations.
Should be able to add that now

**Review by ludeeus — COMMENTED** [review]
✅ symbolic approval as I can not approve my own PR.

**Review by emontnemery — DISMISSED** [review]
Nice 👍

**Review by emontnemery — APPROVED** [review]

**Pull request merged**